conversions/src/Data/Conversions.hs

```diff
@@ -244,8 +244,8 @@ checkedFromIntegral value = guard' (fromIntegral converted == value) converted
 convertEither :: forall b a e . (Conversion (Either e b) a) => a -> Either e b
 convertEither = convert
 
-convertUnsafe :: forall b a e . (Conversion (Either e b) a, Show e) => a -> b
-convertUnsafe = either (error . show) id . convertEither @b @a @e
+convertAssert :: forall b a e . (Conversion (Either e b) a, Show e) => a -> b
+convertAssert = either (error . show) id . convertEither @b @a @e
```

Comment:
@mbj  one small issue with changing to `convertAssert` is that the name or functional signature doesn't give any indication of the fact this function is partial. And I am having a hard time thinking of a better name that conveys both the intention and partiality of this function.

Reply:
Yeah so we need a name with this properties:

* Does not use `Unsafe` as this is overloaded with memory unsafe operations, where this function is not about memory at all.
* Indicates it does impure throws on violation.

I think we may have to go verbose: `convertImpureError` ?

Reply:
@mbj `convertImpure` seems sufficient. `Impure` already implies the possibility of throwing an impure error.